pages/en/contrib/style-guide-es6.md

```diff
@@ -0,0 +1,764 @@
+
+**Good**:
+
+```
+TBD
+```
+
+**Best**:
+
+```js
+const matchesInEquality = testInEquality({ gte: example.between[0] }, value) &&
+    testInEquality({lte: example.between[1]}, value) &&
+    testInEquality({lte: example.between[2]}, value);
+if (matchesInEquality) {
+  handleInEquality();
+}
+```
+
+Still acceptable:
+
+```js
+if (testInEquality({gte: example.between[0]}, value) &&
+    testInEquality({lte: example.between[1]}, value) &&
+    testInEquality({lte: example.between[2]}, value)) {
+  handleInEquality();
+}
+```
+
+**Bad**:
+
+One level of indentation makes it difficult to tell the difference between the condition and the branch body.
+
+```js
+if (testInEquality({gte: example.between[0]}, value) &&
+  testInEquality({lte: example.between[1]}, value) &&
+  testInEquality({lte: example.between[2]}, value)) {
+  handleInEquality();
+}
+```
+
+### Multiline Array
+
+**Good**:
+
+```
+const titles = [
+  {title: 'Title A', subject: 'B'},
+  {title: 'Title Z', subject: 'A'},
+  {title: 'Title M', subject: 'C'},
+  {title: 'Title A', subject: 'A'},
+  {title: 'Title B', subject: 'A'},
+  {title: 'Title C', subject: 'D'},
+];
+```
+
+**Bad**:
+
+```
+const titles = [{title: 'Title A', subject: 'B'},
+                {title: 'Title Z', subject: 'A'},
+                {title: 'Title M', subject: 'C'},
+                {title: 'Title A', subject: 'A'},
+                {title: 'Title B', subject: 'A'},
+                {title: 'Title C', subject: 'D'}];
+```
+
+```
+const titles = [{ title: 'Title A', subject: 'B' },
+  {title: 'Title Z', subject: 'A'},
+  {title: 'Title M', subject: 'C'},
+  {title: 'Title A', subject: 'A'},
+  {title: 'Title B', subject: 'A'},
+  {title: 'Title C', subject: 'D'}];
+```
+
+### Line spacing
+
+In general, group related lines together (with a single empty line in between groups).
+
+```
+if (err) return done(err);
+
+const cat = new Cat();
+cat.eat();
+cat.meow();
+cat.sleep();
+
+return cat;
+```
+
+However, if the method is short (3-5 lines) then just group it all together.
+
+**Good**:
+
+```
+if (err) return done(err);
+expect(result).to...;
+done();
+```
+
+**Bad**:
+
+```
+if (err) return done(err);
+
+expect(result).to...;
+
+done();
+```
+
+## Style guidelines for tests
+
+### Sandbox directories
+
+- All test-related sandbox directories should be inside the `test` dir (ie.
+  `./test/sandbox`)
+- Do not use directories like `/tmp/sandbox` as you will run into
+  permission issues on CI for directories that are not in the project
+
+### Email examples
+
+- All test-related email examples should be of the format `email@example.com`.
+- The `example.com` domain was created to be used for examples in documents, and could be used without prior coordination or asking for permission.
+
+**Good**:
+
+```js
+const validCredentials = {email: `original@example.com`, password: 'bar'}
+```
+
+**Bad**:
+
+```js
+const validCredentials = {email: `updated@bar.com`, password: 'bar'}
+```
+
+### Hooks
+
```

Comment:
It'd be great if you can add an intro to this subsection as you have done with the others.

Reply:
Will do.